code/extensions/LDAPMemberExtension.php

```diff
@@ -165,36 +165,39 @@ public function onBeforeWrite()
         $service->createLDAPUser($this->owner);
     }
 
-    /**
-     * Update the local data with LDAP, and ensure local membership is also set in
-     * LDAP too. This writes into LDAP, provided that feature is enabled.
-     */
     public function onAfterWrite()
     {
+        $this->sync();
+    }
+
+    public function onAfterDelete() {
         $service = Injector::inst()->get('LDAPService');
         if (
             !$service->enabled() ||
-            !$this->owner->config()->update_ldap_from_local ||
+            !$this->owner->config()->delete_users_in_ldap ||
             !$this->owner->GUID
         ) {
             return;
         }
 
-        $service->updateLDAPFromMember($this->owner);
-        $service->updateLDAPGroupsForMember($this->owner);
+        $service->deleteLDAPMember($this->owner);
     }
 
-    public function onAfterDelete() {
+    /**
+     * Update the local data with LDAP, and ensure local membership is also set in
+     * LDAP too. This writes into LDAP, provided that feature is enabled.
+     */
+    public function sync() {
         $service = Injector::inst()->get('LDAPService');
         if (
             !$service->enabled() ||
-            !$this->owner->config()->delete_users_in_ldap ||
+            !$this->owner->config()->update_ldap_from_local ||
```

Comment:
We should probably clarify `update_ldap_from_local` docblock to say it will only automatically sync Member fields, but is unable to sync groups.